lxl-web/src/lib/utils/holdings.ts

```diff
@@ -185,3 +203,163 @@ export function getMyLibsFromHoldings(
 	}
 	return Object.values(result);
 }
+
+export async function getFullHolderData(allHolders: DecoratedHolder[]): Promise<FullHolderBySigel> {
+	const holderBySigel: FullHolderBySigel = {};
+
+	for (const h of allHolders) {
+		const id = h.obj?.['@id'];
+		const libraryRes = await fetch(`${id}?framed=true`, {
+			headers: { Accept: 'application/ld+json' }
+		});
+		const resJson = await libraryRes.json();
+		const libraryMainEntity = resJson['mainEntity'] as FramedData;
+
+		if (libraryMainEntity) {
+			holderBySigel[h.sigel] = libraryMainEntity;
+		}
+	}
+	return holderBySigel;
+}
+
+export async function fetchHoldersIfAbsent(holdersByType: HoldersByType) {
+	const cachedHolders = holdersCache.holders;
+	const allHolders = Object.values(holdersByType).flat();
+	for (const h of allHolders) {
+		const id = h.obj?.['@id'];
+
+		if (h.sigel && cachedHolders && !cachedHolders[h.sigel]) {
+			const response = await fetch(`${id}?framed=true`, {
+				headers: { Accept: 'application/ld+json' }
+			});
+			if (response.ok) {
+				const resJson = await response.json();
+				const libraryMainEntity = resJson['mainEntity'] as FramedData;
+				if (libraryMainEntity) {
+					cachedHolders[h.sigel] = libraryMainEntity;
+				}
+			} else {
+				console.error(`Could not fetch holder data for ${id}`);
+			}
+		}
+	}
+}
+
+export function getItemLinksByBibId(
+	bibIdsByInstanceId: Record<string, BibIdObj>,
+	locale: LocaleCode,
+	displayUtil: DisplayUtil
+): ItemLinksByBibId {
+	const linksByInstanceId: ItemLinksByBibId = {};
+	for (const bibIdObj of Object.values(bibIdsByInstanceId || [])) {
+		const linksForHolder: ItemLinksForHolder = {};
+		bibIdObj.holders?.forEach((sigel) => {
+			if (holdersCache.holders) {
+				const fullHolderData = holdersCache.holders[sigel];
+
+				const ilsPaths = [
+					[BibDb.ils, BibDb.bibIdSearchUri],
+					[BibDb.ils, BibDb.isbnSearchUri],
+					[BibDb.ils, BibDb.issnSearchUri]
+				];
+
+				const lopacPaths = [[BibDb.lopac, BibDb.bibIdSearchUriByLang]];
+
+				let linksToItem = getLinksToItemFor(bibIdObj, fullHolderData, ilsPaths, locale);
+				const lopacLinksItem = getLinksToItemFor(bibIdObj, fullHolderData, lopacPaths, locale);
+
+				const linkTemplateEod = getAtPath(fullHolderData, [BibDb.eodUri], []);
+				if (linkTemplateEod && linkTemplateEod.length !== 0) {
+					linksToItem = [linkTemplateEod.replace(/%BIB_*ID%/, bibIdObj.bibId), ...linksToItem];
```

Comment:
this will only replace the first occurence of %BIB_*ID%
same comment for all `replace`